lblod/frontend-embeddable-notule-editor
# Restructure repository to monorepo

## Overview
This PR includes a restructure to a monorepo setup. Specifically it includes the following changes:
- Introduction of seperate `embeddable-say-editor` and `test-app` workspaces/packages
- Addition of a `Dockerfile` to the `test-app` workspace
- The cypress configuration is now include in the `test-app` workspace. Note cypress does not handle iframes well, so the tests will fail.
- The `test-app` package is implemented using `webpack-dev-server` and supports hot reload.
- ~~Removal of the CI pipelines that build the docker test-app for now.~~
- When a tag is pushed that starts with `v`, such as in a normal release, the module is built and pushed to npm. If the tag begins with `v` but contains `-next.`, such as when using a changeset pre-release, then the module is pushed to npm with the `next` tag, so it will not be installed unless that tag is specifically asked for. This allows for pre-releases. (tested using dummy scripts and non-release tags (those without `v` at the start))
- Seperate the output of ember and webpack in the `ember-build` and `dist` folders
- Only publish the `dist` folder of the `embeddable-say-editor` workspace when publishing to npm

TODOs:
- [x] Add npm lint scripts to root workspace
- [x] Include a backend workspace that contains a `docker-compose` configuration
  - There is no 'backend' but a docker-compose that can be run to publish the test app
- [x] Extend `test-app` to include more configurations
  - We should definitely add more when needed
- [x] Extend release CI pipeline to support npm pre-releases
- [x] Check if `release-it` works as expected
- [x] Check if all config files are in the right place

##### connected issues and PRs:
https://binnenland.atlassian.net/browse/GN-4573?atlOrigin=eyJpIjoiZGE4MGJkZDkwZTY4NDY0ODgzNjU3N2FlNmRkZGM5ODMiLCJwIjoiaiJ9

### How to test/reproduce
- npm i
- run the `npm start` script in the root workspace. This should build the `embeddable-say-editor` in watch mode and start up the `test-app` package
- The test-app should ~~open automatically~~ be accessible on localhost:8080 and should include an iframe.
- Run `npm run docker:build-test` to build a docker image. This image can then be started with `docker run --rm -p 8080:80 <built image hash>`.

### Challenges/uncertainties
- When running the `test-app` package with hot reload, sometimes a `node_modules` error pops up. This does not seem to have any impact on the functionality however.
-  ~~For some reason, ESLint ignores the decorators in the `embeddable-say-editor` workspace.~~ - Fixed by bumping some minor versions of babel and plugins :s

### Checks PR readiness
